pisek/config/task_config.py

```diff
@@ -494,7 +499,13 @@ def convert_yesno(cls, value: str, info: ValidationInfo) -> bool:
     @classmethod
     def convert_auto(cls, value: str, info: ValidationInfo) -> str:
         if value == "@auto":
-            return info.data.get("name", "")
+            value = info.data.get("name", "")
+
+        if value.startswith("_"):
+            raise PydanticCustomError(
+                "invalid_solution_source",
+                f"Solution source must not start with '_'",
+            )
```

Comment:
Is there a reason for this?

Reply:
What in particular you are objecting?

Reply:
Names cannot start with a `_` to prevent collisions with reserved folder names, but I don't see a reason for the source file name not to be able to strart with `_`.

Reply:
We name folders according to sources, not names. As to why enforce same thing is enforced on names, is that it otherwise produces confusing error messages:
```
In section [all_solutions], key 'source' (in config file /home/skipy/dev/pisek/fixtures/sum_cms/cms-defaults):
  Solution source must not start with '_': '@auto'
  ```

Reply:
True, I forgot about that.
Why are we naming the folders according to the source file, anyway?

Reply:
> Why are we naming the folders according to the source file, anyway?

Uups, that probably doesn't work once we allow for same source be different solutions differentiated by arguments. Gonna rewrite that.